### PR TITLE
fix a bug in QuadraticProgram when adding linear cost

### PR DESCRIPTION
**Pull request by hongkai-dai**

### CODE REVIEW

File: drake/solvers/test/testQP.m [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 
 fprintf('****************************************\n');
-fprintf(' min_x (.5*x(1)^2+x(2)^2+x(1)*x(2) + x(1)+2*x(2)), subj to 5<=x(1)<=10;-20<=x(1)+4*x(2)<10\n');
+fprintf(' min_x (.5*x(1)^2+x(2)^2+x(1)*x(2) + x(1)), subj to 5<=x(1)<=10;-20<=x(1)+4*x(2)<10\n');
 fprintf('****************************************\n');
 prog = QuadraticProgram(eye(2),[1;0],[],[],[],[],[],[]);
 prog = prog.addCost(QuadraticConstraint(-inf,inf,[0 1;1 0],zeros(2,1)));
```
Comment (RussTedrake) [inline drake/solvers/test/testQP.m]: the fprintf on the objective still looks wrong to me.

File: drake/solvers/QuadraticProgram.m [hunk 2]
```diff
@@ -130,7 +130,7 @@
       if(cnstr.num_cnstr~= 1)
         error('Drake:QuadraticProgram:UnsupportedConstraint','QuadraticProgram cannot accept LinearConstraint with multiple rows in its objective')
       end
-      obj.f(xind) = obj.f(xind)+A';
+      obj.f(xind) = obj.f(xind)+cnstr.A';
```
Comment (RussTedrake) [inline drake/solvers/QuadraticProgram.m]: this would have thrown a runtime error before, correct?

Comment (hongkai-dai) [inline drake/solvers/QuadraticProgram.m]: Yes, but we have never called `addCost` with a linear cost before. I add a test case that would cover this line.

File: drake/solvers/test/testQP.m [hunk 3]
```diff
@@ -45,6 +45,29 @@
 testSolvers(prog,solvers,1e-3);
 warning(w);
 
+fprintf('****************************************\n');
+fprintf(' min_x (.5*x(1)^2+x(2)^2+x(1)*x(2) + x(1)+2*x(2)), subj to 5<=x(1)<=10;-20<=x(1)+4*x(2)<10\n');
+fprintf('****************************************\n');
+prog = QuadraticProgram(eye(2),[1;0],[],[],[],[],[],[]);
+prog = prog.addCost(QuadraticConstraint(-inf,inf,[0 1;1 0],zeros(2,1)));
+prog = prog.addCost(LinearConstraint(-inf,inf,[0 2]));
```
Comment (RussTedrake) [inline drake/solvers/test/testQP.m]: this fprintf looks wrong, too.  am i misreading?  isn't it just `2*x(1)*x(2) + x(2)`?  maybe with a factor of .5?

Comment (hongkai-dai) [inline drake/solvers/test/testQP.m]: Yes, I will change it. I copied it from the line 26 above, which was wrong in the first place

**Pull request merged**